File: 2161. Partition Array According to Given Pivot.cpp
class Solution {
    public:
        vector<int> pivotArray(vector<int>& nums, int x) {
            vector<int>a;
            vector<int>b;
            vector<int>c;
    
            for(int v:nums)
            {
                if(v==x) c.push_back(v);
                else if(v < x) a.push_back(v);    
                else b.push_back(v);
            }
    
            a.insert(a.end(),c.begin(),c.end());
            a.insert(a.end(),b.begin(),b.end());
            return a;
        }
    };

File: Array/2161. Partition Array According to Given Pivot.py
class Solution:
    def pivotArray(self, nums: List[int], pivot: int) -> List[int]:
        less_val, greater_val, equal_val = [], [], []

        for val in nums:
            if val == pivot: equal_val.append(val)
            elif val < pivot: less_val.append(val)
            else: greater_val.append(val)

        return less_val + equal_val + greater_val